snooty/types.py

```diff
@@ -127,7 +127,7 @@ class BundleConfig:
 @dataclass
 class AssociatedProduct:
     name: str
-    versions: List[str]
+    versions: Optional[List[str]] = field(default_factory=list)
```

Comment:
Optional is `T | None` -- do we actually want to accept None here, or if it's omitted, do we want an empty list?

Reply:
we actually want to remove it completely but the cloud docs already has it, so it will be an intermediate step to make it optional :T (if we do remove it completely, parser will complain about an unknown config)

so in short, we want to accept None here

Reply:
Got it. All else being equal I think I prefer not to have this be Optional at all -- I don't like None's without a reason to have them. But since it sounds like this will never be used by any parser code, I think it's good to merge as-is :D